Server/mods/deathmatch/logic/CStaticFunctionDefinitions.cpp

```diff
@@ -4928,14 +4943,50 @@ bool CStaticFunctionDefinitions::TakeAllWeapons(CElement* pElement)
         CPed* pPed = static_cast<CPed*>(pElement);
         if (pPed->IsSpawned())
         {
-            CBitStream BitStream;
-            m_pPlayerManager->BroadcastOnlyJoined(CElementRPCPacket(pPed, TAKE_ALL_WEAPONS, *BitStream.pBitStream));
+            std::vector<std::tuple<unsigned char, unsigned char, unsigned char>> weapons;
 
             for (unsigned char ucWeaponSlot = 0; ucWeaponSlot < WEAPON_SLOTS; ++ucWeaponSlot)
             {
-                pPed->SetWeaponType(0, ucWeaponSlot);
-                pPed->SetWeaponAmmoInClip(0, ucWeaponSlot);
-                pPed->SetWeaponTotalAmmo(0, ucWeaponSlot);
+                unsigned char ucWeaponID = pPed->GetWeaponType(ucWeaponSlot);
+                unsigned char ucAmmo = pPed->GetWeaponTotalAmmo(ucWeaponSlot);
+
+                if (ucWeaponID > 0)
+                {
+                    CLuaArguments arguments;
+                    arguments.PushNumber(ucWeaponID);
+                    arguments.PushNumber(ucAmmo);
+                    arguments.PushNumber(ucWeaponSlot);
+
+                    bool shouldTake = true;
+                    if (!pPed->CallEvent(IS_PLAYER(pElement) ? "onPlayerWeaponTaken" : "onPedWeaponTaken", arguments))
+                        shouldTake = false;
+
+                    if (shouldTake)
+                    {
+                        weapons.push_back({ucWeaponID, ucAmmo, ucWeaponSlot});
+
+                        pPed->SetWeaponType(0, ucWeaponSlot);
+                        pPed->SetWeaponAmmoInClip(0, ucWeaponSlot);
+                        pPed->SetWeaponTotalAmmo(0, ucWeaponSlot);
+                    }
+                }
+            }
+
+            if (!weapons.empty())
+            {
+                CBitStream    BitStream;
+                std::size_t weaponsTaken = (std::size_t)weapons.size();
```

Comment:
`std::size_t` is platform specific type. When it comes to networking, we should use platform independent types like `std::uint32_t`